internal/memoize/README.md

```diff
@@ -0,0 +1,11 @@
+# Memoize
+
+Memoize allows to cache certain expensive function calls and
+cache the result. The main advantage in Coraza is to memoize
+the regexes when the connects spins up more than one WAF in
+the same process and hence same regexes are being compiled
+over and over.
+
+Currently it is opt-in under the `memoize_regex` build tag
```

Comment:
What about also adding a one-line description in the main readme, under https://github.com/corazawaf/coraza#build-tags?